packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -96,6 +97,68 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
   bool shouldRelayout(MultiChildLayoutDelegate oldDelegate) => false;
 }
 
+class _FloatingActionButtonTransition extends StatefulComponent {
+  _FloatingActionButtonTransition({
+    Key key,
+    this.child
+  }) : super(key: key) {
+    assert(child != null);
+  }
+
+  final Widget child;
+
+  _FloatingActionButtonTransitionState createState() => new _FloatingActionButtonTransitionState();
+}
+
+class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTransition> {
+  final Performance hidePerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  final Performance showPerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  Widget oldChild;
+
+  void initState() {
+    super.initState();
+    showPerformance.play();
+  }
+
+  void dispose() {
+    hidePerformance.stop();
+    showPerformance.stop();
+    super.dispose();
+  }
+
+  void didUpdateConfig(_FloatingActionButtonTransition oldConfig) {
+    if (Widget.canUpdate(oldConfig.child, config.child))
+      return;
+    oldChild = oldConfig.child;
+    showPerformance.progress = 0.0;
+    hidePerformance
+      ..progress = 0.0
+      ..play().then((_) {
+        oldChild = null;
+        showPerformance.play();
+     });
```

Comment:
Rather than using two chained performance, it's better to use a single performance with two intervals.  That way you don't have any problem if there's a jank somewhere else in the system at the moment when you want to switch between the two performances.  Because there's only one clock, all the timings will be consistent.